filament/src/ShadowMap.cpp

```diff
@@ -624,22 +636,22 @@ mat4f ShadowMap::getTextureCoordsMapping() const noexcept {
     return Mf * Mb * Mv * Mt;
 }
 
-math::mat4f ShadowMap::computeVsmLightSpaceMatrix(const math::mat4f& lightSpacePcf, const
-        math::mat4f& Mv, float zfar) noexcept {
+math::mat4f ShadowMap::computeVsmLightSpaceMatrix(const math::mat4f& lightSpacePcf,
+        const math::mat4f& Mv, float zfar) noexcept {
     // The lightSpacePcf matrix transforms coordinates from world space into (u, v, z) coordinates,
-    // where (u, v) are used to access the shadow map, and z is the PCF comparison value between 0
-    // and 1.
-    // For VSM, we want to leave the z coordinate in linear light space, with a few adjustments:
-    //   - We scale the z by zfar, which prevents us from overflowing when we square z in the depth
-    //     shader.
-    //   - The negative sign accounts for our right-handed coordinate system, where -z is forward.
+    // where (u, v) are used to access the shadow map, and z is the (non linear) PCF comparison
+    // value [0, 1].
+    //
+    // For VSM, we want to leave the z coordinate in linear light space, normalized between [0, 1]
+    // (the normalization factor is therefore -1/zfar).
+    //
     // When sampling a VSM shadow map, the shader follows suit, and doesn't divide by w for the z
     // coordinate. See getters.fs.
-    math::mat4f lightSpaceVsm = lightSpacePcf;
-    lightSpaceVsm[0][2] = Mv[0][2] * (-1.0f / std::abs(zfar));
-    lightSpaceVsm[1][2] = Mv[1][2] * (-1.0f / std::abs(zfar));
-    lightSpaceVsm[2][2] = Mv[2][2] * (-1.0f / std::abs(zfar));
-    lightSpaceVsm[3][2] = Mv[3][2] * (-1.0f / std::abs(zfar));
+    math::mat4f lightSpaceVsm{ lightSpacePcf };
```

Comment:
I think we can remove the negative sign here, and also in the shader.

Reply:
I think it'll be less confusing to leave it this way, so we get values between 0 and 1.

Reply:
Yeah, nevermind I was looking at the wrong equation